docs/DdnRelocatableShaderElf.md

```diff
@@ -0,0 +1,205 @@
+
+The two concepts are similar, but there is a minor, and important, distinction.
+An unlinked shader is an LGC concept. LGC will generate an unlinked shader when
+[`m_unlinked`](../llpc/context/llpcPipelineContext.h) in the pipeline context is
+set. When possible, LGC will generate code that can be linked with other
+unlinked shaders and the pipeline state to create the pipeline. The link step is
+supposed to be relatively fast, and it is implemented in the `ElfLinkerImpl`
+class in [ElfLinker.cpp](../lgc/elfLinker/ElfLinker.cpp).
+
+A relocatable shader is an LLPC concept. The flag
+[`enableRelocatableShaderElf`](../include/vkgcDefs.h) in the pipeline options
+causes LLPC to use relocatable shaders when building the pipeline. When building
+with relocatable shaders, LLPC will create a new pipeline state object for each
+shader, stripping out as much as it can from the pipeline state. The new
+pipeline is passed to LGC, where it will be compiled as an unlinked shader. LLPC
+will then link all of those unlinked shaders together to create the pipeline.
+
+The key difference is that unlinked shaders are built with whatever information
+they have while relocatable shaders remove anything that can be handled at link
+time. For example, if LGC is asked to compile an unlinked shader and the
+pipeline state contains the user data nodes, then relocations will not be added
+for the descriptor loads. However, if LLPC is asked to compile a pipeline using
+relocatable shaders, the user data nodes will be present. LLPC will not pass
+that information to LGC when compiling the unlinked shaders, and relocations
+will be used.
+
+# Strategies for handling missing pipeline state
+
+## Relocations
+
+[Relocations in elf files](https://refspecs.linuxbase.org/elf/gabi4+/ch4.reloc.html)
+are the standard way to fill in missing information that is only available when
+linking object files. Relocations are often used to represent the address of a
+symbol to be accessed. Once the address of the symbol has been resolved, the
+linker can fill in the value. Unlinked shaders will create relocations for the
+absolute address of special symbols. These special symbols will not be given an
+actual address to be used, but their address will represent a value to be used.
+
+One example is the descriptor offset relocation. The symbol will have a name of
+the form `doff_<d>_<b>_<t>`, where `<d>` is the descriptor set and `<b>` is the
+binding of the descriptor that we are interested in. The type of the descriptor
+is given by `<t>`. For example, a relocation asking for the absolute address of
+`doff_2_1_s` will be replaced by the offset of the sampler (`s`) in descriptor
+set 2 and binding 1.
+
+This mechanism can be used to replace any 32-bit value that needs to appear in
+the `.text` section of the unlinked shader. The current list of the special
+symbols to which relocations can be applied can be found in
+[AbiUnlinked.h](../lgc/include/lgc/state/AbiUnlinked.h).
+
```

Comment:
Is there a need to discuss 64-bit relocations?

Reply:
We currently do not have any 64-bit relocations.  There are separate relocations that PAL will be accepting where you will want to relocate a 64-bit address, but even those are separated into to 32-bit relocations.  Those are unrelated to unlinked/relocatable shaders.